cli/src/semgrep/config_resolver.py

```diff
@@ -252,18 +252,11 @@ def _load_config_from_local_path(self) -> List[ConfigFile]:
     def is_registry_url(self) -> bool:
         return self._origin == ConfigType.REGISTRY
 
-    def _project_metadata_for_standalone_scan(
-        self, require_repo_name: bool
-    ) -> out.ProjectMetadata:
+    def _project_metadata_for_standalone_scan(self) -> out.ProjectMetadata:
         repo_name = os.environ.get("SEMGREP_REPO_NAME")
 
         if repo_name is None:
-            if require_repo_name:
-                raise SemgrepError(
-                    f"Need to set env var SEMGREP_REPO_NAME to use `--config {self._config_path}`"
-                )
-            else:
-                repo_name = "unknown"
+            repo_name = "unknown"
```

Comment:
@jbergler are we okay with a solution like this? Don't feel to strongly about passing "unknown", would rather pass `None`. 

However that requires interface updates, but don't want to mess with the contract. Thoughts on making `repository` in `ProjectMetadata` interface, optional?